src/main/java/org/mapstruct/intellij/inspection/TargetPropertyMappedMoreThanOnceInspection.java

```diff
@@ -0,0 +1,210 @@
+import com.intellij.openapi.editor.CaretState;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.LogicalPosition;
+import com.intellij.openapi.editor.ScrollType;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.TextEditor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.util.text.Strings;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiAnnotationMemberValue;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiType;
+import org.jetbrains.annotations.NotNull;
+import org.mapstruct.intellij.MapStructBundle;
+import org.mapstruct.intellij.util.MapStructVersion;
+import org.mapstruct.intellij.util.MapstructUtil;
+import org.mapstruct.intellij.util.TargetUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.intellij.codeInsight.AnnotationUtil.getStringAttributeValue;
+import static org.mapstruct.intellij.util.MapstructAnnotationUtils.extractMappingAnnotationsFromMappings;
+import static org.mapstruct.intellij.util.MapstructUtil.MAPPINGS_ANNOTATION_FQN;
+import static org.mapstruct.intellij.util.MapstructUtil.MAPPING_ANNOTATION_FQN;
+import static org.mapstruct.intellij.util.TargetUtils.getTargetType;
+
+/**
+ * @author hduelme
+ */
+public class TargetPropertyMappedMoreThanOnceInspection extends InspectionBase {
+    @NotNull
+    @Override
+    PsiElementVisitor buildVisitorInternal(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new TargetPropertyMappedMoreThanOnceInspection.MyJavaElementVisitor( holder,
+                MapstructUtil.resolveMapStructProjectVersion( holder.getFile() ) );
+    }
+
+    private static class MyJavaElementVisitor extends JavaElementVisitor {
+        private final ProblemsHolder holder;
+        private final MapStructVersion mapStructVersion;
+
+        private MyJavaElementVisitor(ProblemsHolder holder, MapStructVersion mapStructVersion) {
+            this.holder = holder;
+            this.mapStructVersion = mapStructVersion;
+        }
+
+        @Override
+        public void visitMethod(PsiMethod method) {
+            if ( !MapstructUtil.isMapper( method.getContainingClass() ) ) {
+                return;
+            }
+            PsiType targetType = getTargetType( method );
+            if ( targetType == null ) {
+                return;
+            }
+            Map<String, List<PsiElement>> problemMap = new HashMap<>();
+            for (PsiAnnotation psiAnnotation :method.getAnnotations()) {
```

Comment:
```suggestion
            for (PsiAnnotation psiAnnotation : method.getAnnotations()) {
```
super nit-pick, just wondering that checkstyle doesn't check this

Reply:
Good catch. Turns out enhanced for loops checks are disabled by default.

Reply:
Fixed